src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -4528,26 +5851,26 @@ void Instruction::execute() {
         const uint32_t sliceNum =
             (ws + metadata_.operands[0].sme_index.disp) % partition_num;
 
-        std::array<uint32_t, 64> mdata;
-        uint16_t md_size = 0;
+        std::vector<uint32_t> memData;
         uint16_t index = 0;
 
         for (uint16_t x = 0; x < partition_num; x++) {
           uint64_t shifted_active = 1ull << ((x % 16) * 4);
           if (pg[x / 16] & shifted_active) {
-            mdata[md_size] = sourceValues_[x].getAsVector<uint32_t>()[sliceNum];
-            md_size++;
-          } else if (md_size) {
+            memData.push_back(
+                sourceValues_[x].getAsVector<uint32_t>()[sliceNum]);
+          } else if (memData.size() > 0) {
             memoryData_[index] =
-                RegisterValue((char*)mdata.data(), md_size * 4);
-            md_size = 0;
+                RegisterValue((char*)memData.data(), memData.size() * 4);
```

Comment:
`sizeof(uint32_t)` might be better than `4` but not that important. Applied throughout

Reply:
Again, if this is something we want to enforce then it should be its own PR as this occurs extremely often in this file. An instruction's opcode defines the data type used so we know what the multiplicand should be

Reply:
Similar also occurs in instruction_address

Reply:
I mainly say this as the `sizeof` syntax is used in other places throughout this file. It prevents the use of magic numbers so is preferable but would also be consistent. I think going forward this syntax would be beneficial, if we want to change it throughout then its own PR would be sensible

Reply:
No action needed, resolve the conversation when you see it